lib/sdk/remote/utils.js

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+"use strict";
+
+const { Class } = require('../core/heritage');
+const { List, addListItem, removeListItem } = require('../util/list');
+const { emit } = require('../event/core');
+
+function forwardEvents(source, target, extraArg) {
+  let listener = (event, ...args) => {
+    emit(target, event, extraArg, ...args);
+  }
+
+  source.on('*', listener);
+
+  return function detach() {
+    source.off('*', listener);
+  }
+}
```

Comment:
So the code here does the distinction that events emitted to the process/frame objects (or their ports) don't include the process/frame as the first argument whereas the events emitted the the processes/frames objects do. Having since seen that tabs/windows doesn't do this, all the events received the relevant target as the first argument, I think it makes more sense to do that. It's consistent and at least a few times while developing this I made a mistake because of it so I've switched to that which makes pipe work just fine (though I'm a little worried by GC cycles being an issue)